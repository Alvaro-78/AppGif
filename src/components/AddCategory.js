import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = e => {
        setInputValue(e.target.value);
    };

    const handleSubmit = e => {
        e.preventDefault();
        // Recibimos por props el estado del componente padre y con el spred operator
        //  le pasamos el nuevo estado con el value del input.
        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        } else {
        }
    };

    return (
        <>
            <p>{inputValue}</p>
            <form onSubmit={handleSubmit}>
                <div className='input-div'>
                    <input
                        type='text'
                        name='input'
                        value={inputValue}
                        onChange={handleInputChange}
                    ></input>
                </div>
                <br />
                <button type='submit'>AÃ±adir...</button>
            </form>
        </>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
