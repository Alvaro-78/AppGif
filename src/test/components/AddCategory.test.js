import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('test component', () => {
    const setCategories = () => {
        return 'cualquier cosa';
    };
    const wrapper = shallow(<AddCategory setCategories={setCategories} />);

    it('should print addCategory component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should changed input text', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe(value);
    });
});
