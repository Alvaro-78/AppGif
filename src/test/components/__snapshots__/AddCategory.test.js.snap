// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should print addCategory component 1`] = `
<Fragment>
  <p />
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-div"
    >
      <input
        name="input"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <br />
    <button
      type="submit"
    >
      AÃ±adir...
    </button>
  </form>
</Fragment>
`;
